columnar/src/backend/columnar/columnar_cache.c

```diff
@@ -189,6 +213,25 @@ static void EvictCache(uint64 size)
 
 			if (entry->readCount == lastCount)
 			{
+				bool skipCacheEntry = false;
+				ListCell *lc;
+				foreach(lc, ChunkGroupsInUse)
+				{
+					ColumarCacheChunkGroupInUse *chunkGroupInUse =
+						(ColumarCacheChunkGroupInUse *) lfirst(lc);
```

Comment:
why another scan here?  we could add a field to the cache struct and mark it there, rather than an external scan for each eviction.

Reply:
I thought about this - just to extend cache information with additional field, but i expect that this would be much more expensive when memory and execution is compared.

List that is created in PR contains only information about current chunks per index scan - expectation that list will be small. When memory is looked, there is 24 bytes (per list entry) against n * 8 bytes when this will be added as part of cache structure.

Regarding execution, invalidation (remove boolean marking) of previous active chunk should be done each time when new chunk is added.

To put in  real use perspective, issues was discovered when 2 index scans were happening in same process (same cache). Eviction happens, in average, after ~3000 cache entries (there are some outliers that evict after 5000, 10000). 
So when we compare memory it is 48 bytes against ~3000. For invalidation scan it is hard to put into exact  rule, but I expect that it would take more using separate list - which will iterate ~6000 times for eviction.